Add GSoC project related vignette chunks

vi-to opened the pull request:
The aim is to add a vignette like documentation of the coding done during the GSoC.

All the code written is documented in place. However, the object documentation added during the last months is perhaps too verbose and could be overwhelming in the help window. Therefore I would suggest that once a complete vignette is in place and freely accessible we may revise internal docs so to provide main concepts and usages, while pointing to the vignette for more elaborated ones. 

Also, there was a small portion of pre-existent text added by @jaymon0703 . I have tried not to mess with his intentions, meaning that I have not radically changed the structure of the document. 

* In the revisions, we need to adjust plots and code examples which are now trimmed; 
* On the examples, I added code snippets when possible or otherwise left TODOs with indications to add them once other conditions are met (datasets are open sourced, outputs are stored and loaded as opposed to call functions while knitting, etc.). Usage examples are often already provided in the docs or anyways discussed in the corresponding PR, it shouldn't take much to retrieve them here.

There is work in progress to provide an organic view on all the topics involved. It is expected that we will go through revisions to guarantee fluid explanations and practical examples, even after the GSoC.

vi-to committed:
Add vignette text, formulas and some examples

The text mostly comes from docs, which are very verbose.

On the examples I added snippets when possible, otherwise I left TODOs with indications to easily add examples later when other conditions are met.

The AC model example snipped is commented out because I branched off 54_algotca before 'acOptTxns' was merged

vi-to committed:
Merge branch '54_algotca' of git://github.com/braverock/blotter into algoTCA/vignette

This merge is necessary to include AC model examples snippets using 'acOptTxns'. This because the algoTCA/vignette branch was created before the #99 was merged.

braverock commented:
Function documentation should always be complete  without referring to a vignette.  opening a pdf or html vignette isn't possible from a terminal, for instance, but ? is always available.

Vignettes are good for examples, tutorials, and more narrative explanation than is reasonable in documentation.  vignettes also more readily incorporate graphics and tablular output.  math is rendered completely in vignettes in ways which it is not in terminal documentation as well.

All of this is great, but we absolutely should *not* cripple the main function documentation in favor of "moving" things to vignettes.

vi-to commented:
Thank you Brian for clear motivations. I agree with your view to keep a rich documentation there as well, it may be just worth doing some slight adjustments in the future. Time will tell, I am not doing any such change there for now.

vi-to committed:
Adjust plots and code snippets

This commit is for sake of avoiding heavily trimmed or misplaced plots in the document.

Future revisions still need to address proper scales around and prevent margins to cover titles or axes, but in the meanwhile this will make it at least readable.

The same goes for the code snippets to show
